### PR TITLE
Pass run and drive --trace-systrace for iOS

### PR DESCRIPTION
**Pull request by jmagman**
issue https://github.com/flutter/flutter/issues/105253

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by jmagman** [comment]
Closing, @gaaclarke is taking this over.

**Comment by gaaclarke** [comment]
Now that I've debugged the issue I think this is good.  All the other problems are in different repos and/or can be addressed separately.  Thanks @jmagman.

**Comment by gaaclarke** [comment]
(Oh, I'm going to remove the "fixes" annotation too.)

**Review by gaaclarke — CHANGES_REQUESTED** [review]
Please update the documentation:
```diff
--- a/packages/flutter_tools/lib/src/commands/run.dart
+++ b/packages/flutter_tools/lib/src/commands/run.dart
@@ -112,7 +112,8 @@ abstract class RunCommandBase extends FlutterCommand with DeviceBasedDevelopment
       ..addFlag('trace-systrace',
         negatable: false,
         help: 'Enable tracing to the system tracer. This is only useful on '
-              'platforms where such a tracer is available (Android and Fuchsia).',
+              'platforms where such a tracer is available (Android, iOS, '
+              'macOS and Fuchsia).',
       )
       ..addFlag('trace-skia',
         negatable: false,
```

**Review by gaaclarke — APPROVED** [review]
I just went ahead and changed it for you (\me pats himself on the back)

**Comment by jmagman** [comment]
> I just went ahead and changed it for you (\me pats himself on the back)

Thank you :deep bow:
Try the suggestion feature next time, it's handy for that, you can merge it from the GitHub UI
Screen Shot 2022-06-03 at 3 26 00 PM
Screen Shot 2022-06-03 at 3 26 57 PM

**Comment by gaaclarke** [comment]
Yea, I think that may only work on files that already have edits on them though.  Maybe if you jump into the full github editor you can start commenting on files that haven't changed?

**Comment by jmagman** [comment]
> Yea, I think that may only work on files that already have edits on them though.

oh right right right

File: packages/flutter_tools/lib/src/commands/run.dart [hunk 1]
```diff
@@ -112,7 +112,8 @@ abstract class RunCommandBase extends FlutterCommand with DeviceBasedDevelopment
       ..addFlag('trace-systrace',
         negatable: false,
         help: 'Enable tracing to the system tracer. This is only useful on '
-              'platforms where such a tracer is available (Android and Fuchsia).',
+              'platforms where such a tracer is available (Android, iOS, '
+              'macOS and Fuchsia).',
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/commands/run.dart:116]: Is it set up in the engine for macOS?

Comment (gaaclarke) [reply packages/flutter_tools/lib/src/commands/run.dart:116]: Dart supports it, I didn't verify that the engine is piping it down to dart correctly though.  I'm not too familiar with the macos flow.

Comment (jmagman) [reply packages/flutter_tools/lib/src/commands/run.dart:116]: So should this have been changed to document that it works on macOS?

Comment (gaaclarke) [reply packages/flutter_tools/lib/src/commands/run.dart:116]: Yea, macOS is a platform "where such a tracer is available".

**Pull request merged**